web/unit_tests/cases/element_interfaces.js

```diff
@@ -803,6 +805,14 @@ if(typeof InterfaceTests == 'undefined') {
       pair.wrapper.deleteCharsBeforeCaret(2);
       assert.equal(pair.wrapper.getText(), Apple.mixed.substr(0, 1) + Apple.mixed.substr(4), "Error deleting context chars from a mixed SMP string");
       String.kmwEnableSupplementaryPlane(false);
+
+      String.kmwEnableSupplementaryPlane(false);
+      testObj.resetWithText(pair, Apple.normal);
+      testObj.setCaret(pair, 4);
+      pair.wrapper.deleteCharsBeforeCaret(7);
```

Comment:
The essence of the test update:  the error was happening when the `OutputTargets` were told to delete more characters than existed in context.

Text context:  `appl | e`
(`Apple.normal = "apple")

So, the test says to delete 7 characters where 4 exist.  The `e` should remain afterward, while all pre-caret text should be removed.  Before the fix, some of the pre-caret text would remain.

Reply:
:+1: